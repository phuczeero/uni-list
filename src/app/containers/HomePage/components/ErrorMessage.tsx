import React, { useEffect, useRef } from 'react';
import styled from 'styled-components/macro';
import { HiExclamation } from 'react-icons/hi';

import { Button } from 'app/components/Button';

interface Props {
  message: string;
  onClose: () => void;
}

export function ErrorMessage(props: Props) {
  const { message, onClose } = props;
  const btnRef = useRef<HTMLButtonElement>(null!);

  useEffect(() => {
    btnRef.current.focus();
  }, []);

  return (
    <Overlay onClick={onClose}>
      <Wrapper>
        <HiExclamation />
        <Message>{message}</Message>
        <Button ref={btnRef} variant="outline" autoFocus>
          Okay
        </Button>
      </Wrapper>
    </Overlay>
  );
}

const Overlay = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(255, 255, 255, 0.75);
`;

const Wrapper = styled.div`
  padding: ${p => p.theme.space[3]}px;
  font-size: ${p => p.theme.fontSizes[5]}px;
  text-align: center;
`;

const Message = styled.div`
  color: ${p => p.theme.colors.danger};
`;
