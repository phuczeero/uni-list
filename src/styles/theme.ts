const fonts = {
  body:
    '-apple-system, BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji"',
};
const fontSizes = [12, 14, 16, 20, 24, 32, 48, 64, 72];
const space = [0, 4, 8, 16, 32, 64, 128, 256, 512];
const radii = {
  default: 6,
  circle: 99999,
};
const colors = {
  text: '#212529',
  muted: '#dee2e6',
  primary: '#007bff',
  danger: '#dc3545',
  blues: {
    light: '#dcedff',
    dark: '#004896',
  },
};
const sizes = {
  navbar: 64,
  spinner: 32,
};
const shadows = {
  card: '0 0 4px rgba(0, 0, 0, .125)',
};
const breakpoints = ['576px', '768px', '992px', '1200px'];

export const theme = {
  fonts,
  fontSizes,
  space,
  radii,
  colors,
  sizes,
  shadows,
  breakpoints,
};

export type ThemeType = typeof theme;
