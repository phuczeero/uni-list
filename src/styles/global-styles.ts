import { createGlobalStyle } from 'styled-components';
import { ThemeType } from './theme';

export const GlobalStyle = createGlobalStyle<{ theme: ThemeType }>`
  * {
    font-family: ${p => p.theme.fonts.body};
    color: ${p => p.theme.colors.text}
  }
`;
