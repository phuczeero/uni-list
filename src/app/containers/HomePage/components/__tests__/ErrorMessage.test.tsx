import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { ErrorMessage } from '../ErrorMessage';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles/theme';

const renderWithTheme = props =>
  render(
    <ThemeProvider theme={theme}>
      <ErrorMessage {...props} />,
    </ThemeProvider>,
  );

describe('<ErrorMessage />', () => {
  it('should render and match the snapshot', () => {
    const component = renderWithTheme({ message: 'test', onClose: () => {} });
    expect(component.container.firstChild).toMatchSnapshot();
  });

  it('should focus button on mount', () => {
    const component = renderWithTheme({ message: 'test', onClose: () => {} });
    const button = component.getByRole('button');
    expect(button).toHaveFocus();
  });

  it('should handle close button clicked', () => {
    const mock = jest.fn();
    const component = renderWithTheme({ message: 'test', onClose: mock });
    const button = component.getByRole('button');
    fireEvent.click(button);
    expect(mock).toBeCalled();
  });
});
