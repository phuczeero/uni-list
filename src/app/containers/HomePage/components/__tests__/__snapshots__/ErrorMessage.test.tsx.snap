// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> should render and match the snapshot 1`] = `
.c3 {
  outline: none;
  padding: 8px;
  background-color: transparent;
  color: #007bff;
  border-radius: 6px;
  border: 2px solid #007bff;
}

.c3:hover,
.c3:focus {
  background-color: #dcedff;
  border-color: #007bff;
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(255,255,255,0.75);
}

.c1 {
  padding: 16px;
  font-size: 32px;
  text-align: center;
}

.c2 {
  color: #dc3545;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="c2"
    >
      test
    </div>
    <button
      class="c3"
    >
      Okay
    </button>
  </div>
</div>
`;
